metricbeat/autodiscover/builder/hints/metrics.go

```diff
@@ -214,6 +228,16 @@ func (m *metricHints) getHostsWithPort(hints common.MapStr, port int, noPort boo
 	return result, true
 }
 
+// getPortName returns the port name defined in a host hint ie: for ${data.host}:%{data.ports.web} returns web
```

Comment:
The host hint can be an URL like `http://${data.host}:%{data.ports.prometheus}/metrics`, right? Would this method work on this case?
Could you add a test case for a case like this one?

Reply:
That needed some tuning! Thanks for mentioning this case!